civic_scraper/platforms/civic_clerk/site.py

```diff
@@ -0,0 +1,106 @@
+
+        e = {'url': full_asset_url,
+             'asset_name': asset_name,
+             'committee_name': committee_name,
+             'place': None,
+             'state_or_province': None,
+             'asset_type': asset_type,
+             'meeting_date': meeting_datetime.date(),
+             'meeting_time': meeting_datetime.time(),
+             'meeting_id': meeting_id,
+             'scraped_by': f'civic-scraper_{civic_scraper.__version__}',
+             'content_type': 'txt',
+             'content_length': None,
+            }
+        return Asset(**e)
+
+    def get_meeting_id(self, event):
+        link = event.xpath("./td[1]//a")[0]
+        href = link.attrib['href']
+        pattern = '.*?\((?P<id>.*?),.*'
+        match = re.match(pattern, href)
+        return 'civicclerk_{}_{}'.format(self.civicclerk_instance, match.group('id'))
+
+    def get_agenda_items(self, text, parser, session):
+        event_tree = etree.parse(StringIO(text), parser)
+        iframe = event_tree.xpath("//iframe")
+        if iframe:
+            event_frame = iframe[0].attrib['src']
+            event_frame_url = 'https://chaffeecoco.civicclerk.com' + event_frame
+            frame_response = session.get(event_frame_url)
+            frame_tree = etree.parse(StringIO(frame_response.text), parser)
+
+            return frame_tree.xpath("//tr[./td[@class='dx-wrap dxtl dxtl__B0' and not(@colspan)]]")
+        return None
+
+    def scrape(self, download=True):
+        session = Session()
+        response = session.get(self.url)
+
+        parser = etree.HTMLParser()
+        tree = etree.parse(StringIO(response.text), parser)
+
+        # only the first <=10 events are here; pagination to be handled later
+        events = tree.xpath("//table[@id='aspxroundpanelRecent2_ASPxPanel4_grdEventsRecent2_DXMainTable']/tr[@class='dxgvDataRow_CustomThemeModerno']")
+
+        ac = AssetCollection()
+
+        for event in events:
+            committee_name = event.xpath("./td[1]//text()")[1].strip()
+            str_datetime = event.xpath("./td[2]//text()")[0].strip()
+            meeting_datetime = datetime.strptime(str_datetime, '%m/%d/%Y %I:%M %p')
+            meeting_id = self.get_meeting_id(event)
+
+            event_url = f'{self.base_url}/Web/DocumentFrame.aspx?id={meeting_id}&mod=-1&player_tab=-2'
+            event_response = session.get(event_url)
+            agenda_items = self.get_agenda_items(event_response.text, parser, session) if event_response.status_code == 200 else None
+
+            # if event_response.status_code != 200:
+            #     print(event_response.status_code)
+
+            if agenda_items:
+                for item in agenda_items:
+                    link_tr_text = item.xpath("./following-sibling::tr[1]")[0]
+                    link_tr = [(tr.attrib['href'], tr.xpath("./text()")[0]) for tr in link_tr_text.xpath(".//a") if tr.attrib['href'] != '#']
+                    if link_tr:
```

Comment:
don't need this check, since the right thing will happen if link_tr is an empty list